pkg/bpf/bpf_linux.go

```diff
@@ -272,7 +272,11 @@ type bpfAttrObjOp struct {
 
 // ObjPin stores the map's fd in pathname.
 func ObjPin(fd int, pathname string) error {
-	pathStr := syscall.StringBytePtr(pathname)
+	pathStr, err := syscall.BytePtrFromString(pathname)
+	if err != nil {
+		return fmt.Errorf("Unable to convert pathname %q to byte pointer: %v", pathname, err)
```

Comment:
I think that it would be useful to wrap the underlying error (using `%w` instead of `%v`) as it can be of type errno and it can be useful to be able to check that.

Reply:
Good point, will change.

Reply:
I started removing explicit "errno functions" in https://github.com/cilium/cilium/pull/10058 but I think that this strategy of wrapping errno errors should be replicated to ease debugging/checking specific errors.